src/call.go

```diff
@@ -114,12 +134,21 @@ func (c *Call) onDCUnpublish(stop []event.SFUTrackDescription, sdp string) {
 
 		trackLogger.Info("unpublishing track")
 
-		if removedTracksCount := c.Conf.RemoveTracksFromPeerConnectionsByInfo(LocalTrackInfo{
-			StreamID: trackDesc.StreamID,
-			TrackID:  trackDesc.TrackID,
-		}); removedTracksCount == 0 {
-			trackLogger.Info("no tracks to remove")
+		newPublishers := []*Publisher{}
+
+		c.mutex.RLock()
+		for _, publisher := range c.Publishers {
+			if publisher.Matches(trackDesc) {
+				publisher.Stop()
+			} else {
+				newPublishers = append(newPublishers, publisher)
+			}
 		}
+		c.mutex.RUnlock()
+
+		c.mutex.Lock()
```

Comment:
probably better to just obtain a write mutex in the first place unless the stop / append operation is likely to take a long time?